### PR TITLE
Add intermediary pvd write

### PR DESCRIPTION
**Pull request by eralpsahin**
Using `stepTime` global array to keep track of the time of each step.
Signature of WritePVD is changed. This will be a breaking change for other examples as well.

### CODE REVIEW

**Comment by eralpsahin** [comment]
Seems we changed the first parameter of Write functions from `argv[1]` to  `meshFile.c_str()`. The filename of the output. I can fix the conflict according to that after your approval.

**Comment by rhk12** [comment]
@eralpsahin you have conflicts. please fix.

File: src/io/output/WritePVD.cpp [hunk 1]
```diff
@@ -3,7 +3,7 @@
 static void strip_ext(char *);
 
 // Write the pvtu file if you are rank zero and code in parallel
-void WritePVD(const char* FileName, int step,double time){
+void WritePVD(const char* FileName, int step, double* stepTime){
```
Comment (rhk12) [inline src/io/output/WritePVD.cpp]: @eralpsahin if stepTime is global does it need to be sent into array? @enstrophy

File: src/io/output/WritePVD.cpp [hunk 2]
```diff
@@ -30,13 +30,13 @@ void WritePVD(const char* FileName, int step,double time){
 	fprintf(fp,"\t<Collection>\n");
 	for(j = 0; j < step+1; j++){
 		//fprintf(fp,"\t\t<Piece Source=\"%s.vtu.%.4d.%.4d\"/>\n", outfileP2,j,i);
-		fprintf(fp,"\t\t<DataSet timestep=\"%d\" file=\"results/%s.%.04d.pvtu\" />\n",j,outfileP2,j);
+		fprintf(fp,"\t\t<DataSet timestep=\"%f\" file=\"results/%s.%.04d.pvtu\" />\n",stepTime[j],outfileP2,j);
```
Comment (enstrophy) [inline src/io/output/WritePVD.cpp]: time= instead of timestep= ?

File: examples/ex5/ex5.cpp [hunk 3]
```diff
@@ -14,6 +14,7 @@ double CenterCoordinate(int i, int coor);
 
 /* Global Variables/Parameters */
 double Time;
+double *stepTime = (double *)calloc(1000,sizeof(double));
```
Comment (enstrophy) [inline examples/ex5/ex5.cpp]: 1000 may be adhoc, allocate using nPlotSteps.

File: src/io/output/WritePVD.cpp (hunk 1 above)
Comment (enstrophy) [reply src/io/output/WritePVD.cpp]: You need not pass the array.

File: examples/ex5/ex5.cpp [hunk 4]
```diff
@@ -190,6 +191,8 @@ int main(int argc, char **argv) {
       printf("------Plot %d: WriteVTU by rank : %d\n", plot_counter,
              world_rank);
       WriteVTU(argv[1], plot_counter, Time);
+      stepTime[nStep] = Time;
+      WritePVD(argv[1], nStep, stepTime);
```
Comment (enstrophy) [inline examples/ex5/ex5.cpp]: Is a call to WritPVD required here?

File: examples/ex5/ex5.cpp [hunk 5]
```diff
@@ -190,6 +191,8 @@ int main(int argc, char **argv) {
       printf("------Plot %d: WriteVTU by rank : %d\n", plot_counter,
              world_rank);
       WriteVTU(argv[1], plot_counter, Time);
+      stepTime[nStep] = Time;
```
Comment (enstrophy) [inline examples/ex5/ex5.cpp]: Check nStep < nPlotSteps to prevent seg. fault

File: examples/ex5/ex5.cpp (hunk 3 above)
Comment (rhk12) [reply examples/ex5/ex5.cpp]: good point.

File: examples/ex5/ex5.cpp (hunk 5 above)
Comment (rhk12) [reply examples/ex5/ex5.cpp]: agree

File: src/io/output/WritePVD.cpp (hunk 2 above)
Comment (eralpsahin) [reply src/io/output/WritePVD.cpp]: As far as I know, `DataSet` tag supports timestep attribute not time attribute.

File: examples/ex5/ex5.cpp (hunk 4 above)
Comment (eralpsahin) [reply examples/ex5/ex5.cpp]: We want to write to PVD after each plot, not only in the end. If there is somewhere else we might prefer writing PVD, could you let me know?

**Comment by rhk12** [comment]
@eralpsahin I get this error when I try to build on windows. 

image

**Comment by rhk12** [comment]
@eralpsahin bump. see comments above.

**Comment by rhk12** [comment]
@eralpsahin is there a way to update your pull request here with most recent develop branch? It looks like you are missing some recent changes. For example line 64 in ex5.cpp should have a line above with conditions sent in from input.json.

**Comment by eralpsahin** [comment]
@rhk12 I am not sure why github did not show conflict without the recent changes but I merged develop again, it should not be missing anymore.

**Comment by rhk12** [comment]
@eralpsahin I think you are still missing updates to input.json

**Comment by rhk12** [comment]
I did see updates to ex5.cpp so that looks good.

**Comment by rhk12** [comment]
nevermind

**Pull request merged**